templates/shared/runtime/bin/set-nvidia-clocks

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+# nvidia-smi is required for this script.
+if ! nvidia-smi -q > /tmp/nvidia-smi-check; then
+  echo >&2 "ERROR: nvidia-smi check failed!"
+  cat /tmp/nvidia-smi-check
+  exit 0
+fi
+
+# it's generally recommended to manually set clocks using max performance in
+# order to get predictable performance.
+# see: https://developer.nvidia.com/blog/advanced-api-performance-setstablepowerstate/
+# see: https://developer.nvidia.com/blog/increase-performance-gpu-boost-k80-autoboost/
+
+# persist device power states so that you dont incur long start up delay when
+# initializing new contexts on a CUDA GPU.
+sudo nvidia-smi --persistence-mode=1
+
+# query the highest speed for both memory and graphics GPU clocks
+# NOTE: su permissions are not required for queries
+MEMORY_CLOCK=$(nvidia-smi --query-supported-clocks=memory --format=csv,noheader,nounits | head -n 1 | tr -d '\n')
```

Comment:
don't need `sudo` here?

Reply:
nope, only the mutating operations require root 👍
omitting `sudo` always gives a nice little cut in execution time :)

Reply:
✅